### PR TITLE
[DOCS] Separate and reformat synced flush API docs

### PR DESCRIPTION
**Pull request by jrodewig**
## Changes
- Separates and reformats the synced flush API docs
- Removes synced flush section from the flush API docs

The synced flush API docs are formatted to align with the new [API reference template](https://github.com/elastic/docs/blob/master/shared/api-ref-ex.asciidoc).

Relates to #43765.

### CODE REVIEW

**Comment by jrodewig** [comment]
@elasticmachine run elasticsearch-ci/2
@elasticmachine run elasticsearch-ci/bwc

**Comment by jrodewig** [comment]
@elasticmachine test this please

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Review by szabosteve — APPROVED** [review]
LGTM

File: docs/reference/indices/synced-flush.asciidoc [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+{es} may trigger an automatic flush of a shard at any time if there are
+unflushed operations in the shard's translog. In practice this means that one
+should consider any indexing operation on an index as having removed its
+`sync_id` markers.
+
+
+[[synced-flush-using-api]]
+===== Use the synced flush API
+
+Use the synced flush API to manually initiate a synced flush.
```
Comment (debadair) [inline docs/reference/indices/synced-flush.asciidoc]: Should this be at the beginning of the description section?

**Review by debadair — APPROVED** [review]
Left a couple minor comments, but otherwise LGTM.

File: docs/reference/indices/synced-flush.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/indices/synced-flush.asciidoc]: Addressed with acda611.

File: client/rest-high-level/src/main/java/org/elasticsearch/client/IndicesClient.java [hunk 2]
```diff
@@ -594,7 +594,7 @@ public Cancellable flushAsync(FlushRequest flushRequest, RequestOptions options,
 
     /**
      * Initiate a synced flush manually using the synced flush API.
-     * See <a href="https://www.elastic.co/guide/en/elasticsearch/reference/master/indices-flush.html#synced-flush-api">
+     * See <a href="https://www.elastic.co/guide/en/elasticsearch/reference/master/indices-synced-flush-api.html">
```
[the remark this answers is not in the document]
Comment (jrodewig) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/IndicesClient.java]: These are in a `.java` file so attributes won't work here.

**Pull request merged**